crates/cast/src/starknet_commands/account/add.rs

```diff
@@ -1,33 +1,50 @@
 use crate::starknet_commands::account::{prepare_account_json, write_account_to_accounts_file};
-use anyhow::{ensure, Result};
+use anyhow::{anyhow, bail, ensure, Result};
 use camino::Utf8PathBuf;
 use cast::helpers::response_structs::AccountAddResponse;
 use cast::helpers::scarb_utils::CastConfig;
+use cast::{get_keystore_password, parse_number};
 use clap::Args;
 use starknet::core::types::FieldElement;
+use starknet::core::utils::get_contract_address;
 use starknet::signers::SigningKey;
+use std::fs;
 
-#[derive(Args, Debug)]
+#[derive(Args, Debug, Default)]
 #[command(about = "Add an account to the accounts file")]
 pub struct Add {
-    /// Address of the account
+    /// Name of the account to be added
     #[clap(short, long)]
+    pub name: String,
+
+    /// Address of the account
+    #[clap(
+        short,
+        long,
+        default_value = "0",
+        required_unless_present = "from_keystore"
```

Comment:
The `default_value` needs to be present for `required_unless_present` to work for some reason
(it doesn't change anything though)